rfcs/0000-ci-all-nix-prs.md

```diff
@@ -0,0 +1,80 @@
+---
+feature: ci-all-nix-prs
+start-date: 2019-10-30
+author: John Ericson (@Ericson2314)
+co-authors: (find a buddy later to help our with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Build all Nix PRs in CI.
+Do not merge any PR until it passes CI.
+
+# Motivation
+[motivation]: #motivation
+
+There is a [famous blog post](blog-post) about how everyone is sloppy and doing CI wrong.
+This isn't just bad for releasing software smoothly, but also increases the burden for new contributors because it is harder to judge the correctness of PRs at a glance (is it broken? Did I break it?).
+I personally find it harder to contribute, I have to worry about double checking all my work on platforms I don't have as-easy access to, like Darwin.
+
+We cannot yet do this for all of Nixpkgs, sadly, due to resource limits.
+But, there is no reason we cannot do it for Nix itself.
+
+# Detailed design
+[design]: #detailed-design
+
+Optional first step: we can set up OfBorg to build all PRs.
+
+Set up Hydra declarative jobsets to build all approved Nix PRs.
+This might involve extending Hydra somewhat.
+Those with merge access should be instructed not to merge a PR until CI passes.
+Merge master into PRs or rebase before merge as a crude stop-gap to avoid master becoming an untested tree due to a merge commit.
+
+If Hydra gains the ability to keep master always working obviating the manual steps above beyond the PR jobs, use that ability.
+
+If a new CI is used, ensure that is also keeps master in an always-building state.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+More process to follow.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Merely build all PRs with OfBorg.
+   This is still far better than the status quo, but has the disadvantage that master must still be rebuilt as OfBorg and Hydra do not share a cache.
+
+2. Merely build all approved PRs, and maintainers are still allowed to merge broken ones / not take care to avoid untested merge commits.
```

Comment:
```suggestion
2. Merely build all approved PRs with Hydra, but maintainers are still allowed to merge broken ones / not take care to avoid untested merge commits.
```